File: server/src/ProxyMity.Domain/Entities/Group.cs
namespace ProxyMity.Domain.Entities;

public class Group
{
    public required Ulid Id { get; set; }

    public required string Name { get; set; }

    public string? Description { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public Ulid CreatedBy { get; set; }

    public static Group Create(Ulid CreatedBy, string name, string? description = null)
    {
        return new Group()
        {
            Id = Ulid.NewUlid(),
            Name = name,
            Description = description,
            CreatedAt = DateTime.UtcNow,
            CreatedBy = CreatedBy,
        };
    }
}

File: Proxymity-Chat-Service-Go/internal/domain/entities/group.go
package entities

import (
	"time"

	"github.com/google/uuid"
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Group struct {
	ObjectId    primitive.ObjectID `bson:"_id,omitempty" json:"objectId"`
	Id          uuid.UUID          `bson:"id" json:"id"`
	Name        string             `bson:"name" json:"name"`
	Description string             `bson:"description" json:"description"`
	CreatedAt   time.Time          `bson:"createdAt" json:"createdAt"`
	UpdatedAt   *time.Time         `bson:"updatedAt" json:"updatedAt"`
	CreatedBy   uuid.UUID          `bson:"createdBy" json:"createdBy"`
}

func NewGroup(name, description string, createdBy uuid.UUID) *Group {
	return &Group{
		Id:          uuid.New(),
		Name:        name,
		Description: description,
		CreatedAt:   time.Now(),
		CreatedBy:   createdBy,
	}
}
